src/test/scala/com/microsoft/hyperspace/index/IndexLogEntryTest.scala

```diff
@@ -139,12 +141,144 @@ class IndexLogEntryTest extends SparkFunSuite {
       Map())
     expected.state = "ACTIVE"
     expected.timestamp = 1578818514080L
 
+    val actual = JsonUtils.fromJson[IndexLogEntry](jsonString)
     assert(actual.equals(expected))
   }
+
+  test("Test files api lists all files from Content object.") {
+    withTempPath { p =>
+      // Prepare some files and directories.
+      Files.createDirectories(Paths.get(p.getAbsolutePath))
+      val dir = Files.createDirectories(Paths.get(p.getAbsolutePath))
+      Files.createTempFile(dir, "f1", "")
+      Files.createTempFile(dir, "f2", "")
+      Files.createTempFile(dir, "f3", "")
+
+      val dirPath = PathUtils.makeAbsolute(dir.toString)
+      val fs = dirPath.getFileSystem(new Configuration)
+      val content = Content.fromDirectory(dirPath)
+
+      val expected = fs.listStatus(dirPath).map(_.getPath).toSet
+      val actual = content.files.toSet
+      assert(actual.equals(expected))
+    }
+  }
+
+  test("Test files api lists all files recursively from Content object.") {
+    withTempPath { p =>
+      // Prepare some files and directories. This time, we make multi-level directory tree.
+      Files.createDirectories(Paths.get(p.getAbsolutePath))
+      val dir = Files.createDirectories(Paths.get(p.getAbsolutePath))
+      Files.createTempFile(dir, "f1", "")
+      Files.createTempFile(dir, "f2", "")
+      Files.createTempFile(dir, "f3", "")
+      val d1 = Files.createTempDirectory(dir, "d1")
+      val d1Path = PathUtils.makeAbsolute(d1.toString)
+      Files.createTempFile(d1, "f4", "")
+
+      val dirPath = PathUtils.makeAbsolute(dir.toString)
+      val fs = dirPath.getFileSystem(new Configuration)
+      val content = Content.fromDirectory(dirPath)
+
+      val expected =
+        (fs.listStatus(dirPath).filter(_.isFile) ++ fs.listStatus(d1Path)).map(_.getPath).toSet
+
+      val actual = content.files.toSet
+      assert(actual.equals(expected))
+    }
+  }
+
+  test("Test fromPath api creates the correct Content object.") {
+    withTempPath { p =>
+      // Prepare some files and directories.
+      Files.createDirectories(Paths.get(p.getAbsolutePath))
+      val dir = Files.createDirectories(Paths.get(p.getAbsolutePath))
+      Files.createTempFile(dir, "f1", "")
+      Files.createTempFile(dir, "f2", "")
+      Files.createTempFile(dir, "f3", "")
+
+      // Create expected Content object.
+      val dirPath = PathUtils.makeAbsolute(dir.toString)
+      val fs = dirPath.getFileSystem(new Configuration)
+      val fileInfos = fs.listStatus(dirPath).map(FileInfo(_))
+      val leafDir = Directory(dirPath.getName, fileInfos)
+      val expected = {
+        val rooDirectory = TestUtils.splitPath(dirPath.getParent).foldLeft(leafDir) {
+          (accum, name) =>
+            Directory(name, Seq(), Seq(accum))
+        }
+
+        Content(rooDirectory, NoOpFingerprint())
+      }
+
+      // Create actual Content object.
+      val actual = Content.fromDirectory(dirPath)
```

Comment:
I think you should also pass `leafDir` and test files under `dirPath` is not included.

Reply:
How about throwIfExists and path filter?

Reply:
thanks @imback82 , added